src/rules/commentFormatRule.ts

```diff
@@ -35,24 +43,60 @@ export class Rule extends Lint.Rules.AbstractRule {
             * \`"check-lowercase"\` requires that the first non-whitespace character of a comment must be lowercase, if applicable.
-            * \`"check-uppercase"\` requires that the first non-whitespace character of a comment must be uppercase, if applicable.`,
+            * \`"check-uppercase"\` requires that the first non-whitespace character of a comment must be uppercase, if applicable.
+            
+            Exceptions to \`"check-lowercase"\` or \`"check-uppercase"\` can be managed with object that may be passed as last argument.
+            
+            One of two options can be provided in this object:
+                
+                * \`"ignoreWords"\`  - array of strings - words that will be ignored at the beginning of the comment.
+                * \`"ignorePattern"\` - string - RegExp pattern that will be ignored at the beginning of the comment.
+            `,
         options: {
             type: "array",
             items: {
-                type: "string",
-                enum: ["check-space", "check-lowercase", "check-uppercase"],
+                anyOf: [
+                    {
+                        type: "string",
+                        enum: [
+                            "check-space",
+                            "check-lowercase",
+                            "check-uppercase",
+                        ],
+                    },
+                    {
+                        type: "object",
+                        properties: {
+                            ignoreWords: {
+                                type: "array",
+                                items: {
+                                    type: "string",
+                                },
+                            },
+                            ignorePattern: {
+                                type: "string",
+                            },
+                        },
+                        minProperties: 1,
+                        maxProperties: 1,
+                    },
+                ],
             },
             minLength: 1,
-            maxLength: 3,
+            maxLength: 4,
         },
-        optionExamples: ['[true, "check-space", "check-lowercase"]'],
+        optionExamples: [
+            '[true, "check-space", "check-uppercase"]',
+            '[true, "check-lowercase", {"ignoreWords": ["TODO", "HACK"]}]',
+            '[true, "check-lowercase", {"ignorePattern": "STD\\w{2,3}\\b"}]',
+        ],
         type: "style",
         typescriptOnly: false,
     };
     /* tslint:enable:object-literal-sort-keys */
 
-    public static LOWERCASE_FAILURE = "comment must start with lowercase letter";
-    public static UPPERCASE_FAILURE = "comment must start with uppercase letter";
+    public static LOWERCASE_FAILURE = "comment must start with lowercase letter, word from exceptions list or exceptions pattern";
```

Comment:
3 different messages are ok.

I'd avoid using the word "exceptions" since it could confuse people with the kind of exception you throw. Something like `Comment must start with a lowercase letter, "TODO", or "HACK"`